File: src/types/app-state.ts
export type DeviceRole = "POS" | "KIOSK" | "QUEUE" | "KDS";

export type SyncStatus = "IDLE" | "SYNCING" | "DONE" | "FAILED";

export interface AppState {
  tenant_domain: string | null;
  access_token: string | null;
  selected_location_name?: string | null;
  order_mode_names?: string[];
  selected_location_id: string | null;
  brand_id: string | null;
  order_mode_ids: string[];
  theme?: string | null;
  language?: string | null;
  direction?: string | null;

  selected_order_mode_id?: string | null;
  selected_order_mode_name?: string | null;

  device_role: DeviceRole | null;
  sync_status: SyncStatus | null;

  ws_server_mode?: number | null;
  ws_server_url?: string | null;
  setup_code?:string|null
}

File: src-tauri/src/db/models/app_state.rs


#[derive(Debug, serde::Serialize)]
pub struct AppState {
    pub tenant_domain: Option<String>,
    pub access_token: Option<String>,
    pub selected_location_id: Option<String>,
    pub selected_location_name: Option<String>,
    pub brand_id: Option<String>,
    pub order_mode_ids: Option<Vec<String>>,
    pub order_mode_names: Option<Vec<String>>,
    pub selected_order_mode_id: Option<String>,
    pub selected_order_mode_name: Option<String>,
    pub device_role: Option<String>,
    pub sync_status: Option<String>,
    pub theme: Option<String>,
    pub language: Option<String>,
    pub kds_view_mode: Option<String>,
    pub kds_settings: Option<String>,
    pub ws_server_mode: Option<i32>,
    pub ws_server_url: Option<String>,
    pub setup_code: Option<String>,
    pub logo_url: Option<String>,
}
